doc/developers/advanced_installation.rst

```diff
@@ -44,6 +44,20 @@ basis to help users test bleeding edge features or bug fixes::
 Building from source
 =====================
 
+In the vast majority of cases, building scikit-learn for development purposes
+can be done with::
+
+    pip install numpy scipy cython pytest flake8
```

Comment:
joblib?

Reply:
I find it also kinda necessary to have sphinx and sphinx-gallery for development.

Reply:
Good point for joblib.
For the doc dependencies, I had them before, but reverted (https://github.com/scikit-learn/scikit-learn/pull/13961/files#r288462944)

I have no strong opinion

Reply:
Joblib will be installed automatically when installing scikit-learn.

Reply:
with the same argument we should also remove numpy and scipy. We have numpy, scipy, and joblib as requirements in setup.py.

Reply:
I removed them. I'm not sure about removing numpy though.